tests/testsuite/update.rs

```diff
@@ -1637,3 +1638,220 @@ fn update_with_missing_feature() {
+        .alternative(true)
+        .publish();
+
+    p.cargo("update --dry-run --breaking")
+        .with_status(1)
+        .with_stderr_contains("[ERROR] unexpected argument '--breaking' found")
+        .run();
+
+    p.cargo("update --breaking")
+        .with_status(1)
+        .with_stderr_contains("[ERROR] unexpected argument '--breaking' found")
+        .run();
+
+    p.cargo("update")
+        .with_stderr(
+            "\
+[UPDATING] `alternative` index
+[UPDATING] `dummy-registry` index
+[LOCKING] 10 packages to latest compatible versions
+[UPDATING] alternative-1 v1.0.0 (registry `alternative`) -> v1.0.1 (latest: v2.0.0)
+[UPDATING] alternative-2 v1.0.0 (registry `alternative`) -> v1.0.1 (latest: v2.0.0)
+[UPDATING] compatible v1.0.0 -> v1.0.1
+[UPDATING] incompatible v1.0.0 -> v1.0.1 (latest: v2.0.0)
+[UPDATING] less-than v1.0.0 -> v2.0.0
+[REMOVING] multiple v1.0.0
+[REMOVING] multiple v2.0.0
+[ADDING] multiple v1.0.1 (latest: v3.0.0)
+[ADDING] multiple v2.0.1 (latest: v3.0.0)
+[UPDATING] pinned v1.0.0 -> v1.0.1 (latest: v2.0.0)
+[UPDATING] renamed-from v1.0.0 -> v1.0.1 (latest: v2.0.0)
+[UPDATING] ws v1.0.0 -> v1.0.1 (latest: v2.0.0)
+",
+        )
+        .run();
+}
+
+#[cargo_test]
+fn update_breaking_specific_packages() {
+    Package::new("just-foo", "1.0.0").publish();
+    Package::new("just-bar", "1.0.0").publish();
+    Package::new("shared", "1.0.0").publish();
+    Package::new("ws", "1.0.0").publish();
+
+    let p = project()
+        .file(
+            "Cargo.toml",
+            r#"
+            [workspace]
+            members = ["foo", "bar"]
+
+            [workspace.dependencies]
+            ws = "1.0"
+        "#,
+        )
+        .file(
+            "foo/Cargo.toml",
+            r#"
+                [package]
+                name = "foo"
+                version = "0.0.1"
+                edition = "2015"
+                authors = []
+
+                [dependencies]
+                just-foo = "1.0"
+                shared = "1.0"
+                ws.workspace = true
+            "#,
+        )
+        .file("foo/src/lib.rs", "")
+        .file(
+            "bar/Cargo.toml",
+            r#"
+                [package]
+                name = "bar"
+                version = "0.0.1"
+                edition = "2015"
+                authors = []
+
+                [dependencies]
+                just-bar = "1.0"
+                shared = "1.0"
+                ws.workspace = true
+        "#,
+        )
+        .file("bar/src/lib.rs", "")
+        .build();
+
+    p.cargo("generate-lockfile").run();
+
+    Package::new("just-foo", "1.0.1").publish();
+    Package::new("just-bar", "1.0.1").publish();
+    Package::new("shared", "1.0.1").publish();
+    Package::new("ws", "1.0.1").publish();
+
+    Package::new("just-foo", "2.0.0").publish();
+    Package::new("just-bar", "2.0.0").publish();
+    Package::new("shared", "2.0.0").publish();
+    Package::new("ws", "2.0.0").publish();
```

Comment:
A test case for `update --breaking just-foo shared ws` that we need to cover is if a breaking version depends on a newer, compatible version of a package.  The upgrade of that transitive dependency should still go through.

Reply:
Added.